cmake/modules/boards.cmake

```diff
@@ -115,7 +115,7 @@ Hints:
 endforeach()
 
 if((HWMv1 AND NOT EXISTS ${BOARD_DIR}/${BOARD}_defconfig)
-   OR (HWMv2 AND NOT EXISTS ${BOARD_DIR}))
+   OR (HWMv2 AND NOT EXISTS ${BOARD_DIR}/board.yml))
   message(WARNING "BOARD_DIR: ${BOARD_DIR} has been moved or deleted. "
                   "Trying to find new location."
```

Comment:
Not a blocker, but we should also handle the case where this `board.yml` still exists, but `${BOARD}` is no longer in it. Hence, we'd need to redo the full search to find the board if it was moved to a different yaml.

Reply:
> Not a blocker, but we should also handle the case where this board.yml still exists, but ${BOARD} is no longer in it.

that is handle, but is handled inside `list_boards.py`.
So in case board is removed, then CMake will fail with a board not valid.
If board is renamed (and properly deprecated in `deprecated.cmake`, then the new board name will be used.

Reply:
I meant the case where the board has been moved, not removed or renamed. So far, this is caught if the cached `${BOARD_DIR}/board.yml` doesn't exist, but imagine moving another `board.yml` in its place, with different boards.

Reply:
Should place a watch on that file for an update you mean? I think that's a reasonable request

Reply:
> but imagine moving another board.yml in its place, with different boards.

so if board folder is named `foo` containing the board `foo` then you would move `foo` board to another place and instead move / update `board.yml` with the content of board `bar` without also renaming the board folder ?

Except for common names for `qemu` boards, then I don't see this as a very likely situation, although it shouldn't be hard to handle this use-case as should just use the board_list output failure, and re-run with no board dir arg.

Reply:
It's a real possibility that there is a folder containing multiple boards and one of those boards is removed entirely or moved somewhere else, thus the boards.yml file is updated, the old location for the board has changed. This is not some rare occurrence

Reply:
> and one of those boards is removed entirely

Which is handled, cause you get an error stating the removed board does not exists.

> moved somewhere else

and in this case the error of the non-existing board this is imprecise.
A user will need to do a pristine build to pickup the new location.

> This is not some rare occurrence

Guess it depends on your definition of rare.
And it's not like it isn't handled, if you read above.

Reply:
Improved error message added for this situation.